packages/server/validators/users.js

```diff
@@ -50,7 +51,113 @@ const userIdBodyValidator = [
     }),
 ];
 
+const onboardingValidator = [
+  // User ID Check: Must Exist, and be new onboarding user.
+  param("userId")
+    .notEmpty()
+    .withMessage("User ID cannot be empty")
+    .blacklist("<>")
+    .isUUID()
+    .withMessage("Invalid user ID provided")
+    .bail()
+
+    // check if user exists and hasn't completed onboarding
+    .custom(async (value) => {
+      try {
+        const user = await UserController.getUser(value);
+        if (user.isNew === false) {
+          throw new Error("User already onboarded!");
+        }
+      } catch (error) {
+        if (error.message === "User already onboarded!") {
+          throw new Error("User already onboarded!");
+        } else {
+          throw new Error("User not found!");
+        }
+      }
+    }),
+
+  // First Name Check: exist, be between 1 and 50 chars
+  body("firstName", "Invalid firstName.")
+    .exists({ checkFalsy: true })
+    .withMessage("firstName can't be null or empty.")
+    .blacklist("<>")
+    .trim()
+    .isLength({ min: 1, max: 50 })
+    .withMessage("firstName length must be between 1 to 255 characters.")
+    .isAlpha()
+    .withMessage("firstName should be Alphabetic characters."),
+
+  // Last Name Check: exist, be between 1 and 50 chars
+  body("lastName", "Invalid lastName.")
+    .exists({ checkFalsy: true })
+    .withMessage("firstName can't be null or empty.")
+    .blacklist("<>")
+    .trim()
+    .isLength({ min: 1, max: 50 })
+    .withMessage("lastName length must be between 1 to 50 characters."),
+
+  // Pronoun Check: exist, be between 1 and 25 chars, and inside the allowed array.
+  body("pronouns", "Invalid pronoun.")
+    .exists({ checkFalsy: true })
+    .withMessage("pronoun can't be null or empty.")
+    .blacklist("<>")
+    .trim()
+    .isLength({ min: 1, max: 25 })
+    .withMessage("pronoun length must be between 1 to 25 characters.")
+    .isIn(ALLOWED_PRONOUN)
+    .withMessage(
+      "pronouns must be one of the allowed values: " +
+        ALLOWED_PRONOUN.join(", "),
+    ),
+
+  // Major Check: exist, be between 1 and 50 chars
+  body("major", "Invalid major.")
+    .exists({ checkFalsy: true })
+    .withMessage("major can't be null or empty.")
+    .blacklist("<>")
+    .trim()
+    .isLength({ min: 1, max: 25 })
+    .withMessage("Major length must be between 1 to 25 characters."),
```

Comment:
This is set to up to 100 characters in the Prisma schema, modify to fit 100 characters instead.
```
major          String?         @db.VarChar(100)
```